x-pack/plugins/fleet/common/services/is_agent_upgradeable.ts

```diff
@@ -42,6 +43,62 @@ export function isAgentUpgradeable(
   return isAgentVersionLessThanLatest(agentVersion, latestAgentVersion);
 }
 
+// Based on the previous, returns a detailed message explaining why the agent is not upgradeable
+export const getNotUpgradeableMessage = (
+  agent: Agent,
+  latestAgentVersion?: string,
+  versionToUpgrade?: string
+) => {
+  let agentVersion: string;
+  if (typeof agent?.local_metadata?.elastic?.agent?.version === 'string') {
+    agentVersion = agent.local_metadata.elastic.agent.version;
+  } else {
+    return `agent version is missing.`;
```

Comment:
It'd be great to move these to string constants so we can reference the same error in multiple places.